[Bug]: Failed to create meal from diary entries.

## Important: Log Requirement for Troubleshooting

To effectively troubleshoot and resolve issues, it is crucial to provide relevant server and/or browser console logs that replicate the problem. Issues submitted without adequate logs may be difficult to diagnose and are at risk of being automatically closed.

## Bug Description

When using the "Create Meal from Diary" modal, submitting fails with "Failed to create meal from diary entries."

## Steps to Reproduce

Steps to reproduce the behavior:
1. Go to Breakfast
2. Click on (+) (Save as new meal)
3. Submit modal
4. See error "Failed to create meal from diary entries."

## Expected Behavior

Meal should get created

## Actual Behavior

<img width="407" height="113" alt="Image" src="https://github.com/user-attachments/assets/b8708e1e-07c4-4c2b-8279-e891116c3806" />

## Environment

- **SparkyFitness Version:** v0.16.3
- **Running Env:** Docker

## Payload

```json
{"date":"2026-01-03","mealType":"breakfast","mealName":"Breakfast - 2026-01-03","description":"","isPublic":false}
```

## Browser Console Log
```
API Call: Error response from /api/meals/create-meal-from-diary: 
{
  "status": 500,
  "errorData": {
    "error": "null value in column \"serving_size\" of relation \"meals\" violates not-null constraint"
  }
}
ConvertToMealDialog: Error creating meal: Error: null value in column "serving_size" of relation "meals" violates not-null constraint
```

## Container Log for SparkyFitness_Server

Please provide the container logs for the `sparkyfitness_server` service. You can usually obtain these logs using `docker logs sparkyfitness_server` or by checking your container orchestration platform's logging interface.

```
sparkyfitness-server-1  | [2026-01-03T17:32:47.222Z] [ERROR] Error creating meal from diary entries: error: null value in column "serving_size" of relation "meals" violates not-null constraint
sparkyfitness-server-1  |     at /app/SparkyFitnessServer/node_modules/pg/lib/client.js:545:17
sparkyfitness-server-1  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
sparkyfitness-server-1  |     at async Object.createMeal (/app/SparkyFitnessServer/models/mealRepository.js:12:24)
sparkyfitness-server-1  |     at async Object.createMealFromDiaryEntries (/app/SparkyFitnessServer/services/mealService.js:416:21)
sparkyfitness-server-1  |     at async /app/SparkyFitnessServer/routes/mealRoutes.js:242:21 {
sparkyfitness-server-1  |   length: 176,
sparkyfitness-server-1  |   severity: 'ERROR',
sparkyfitness-server-1  |   code: '23502',
sparkyfitness-server-1  |   detail: undefined,
sparkyfitness-server-1  |   hint: undefined,
sparkyfitness-server-1  |   position: undefined,
sparkyfitness-server-1  |   internalPosition: undefined,
sparkyfitness-server-1  |   internalQuery: undefined,
sparkyfitness-server-1  |   where: undefined,
sparkyfitness-server-1  |   schema: 'public',
sparkyfitness-server-1  |   table: 'meals',
sparkyfitness-server-1  |   column: 'serving_size',
sparkyfitness-server-1  |   dataType: undefined,
sparkyfitness-server-1  |   constraint: undefined,
sparkyfitness-server-1  |   file: 'execMain.c',
sparkyfitness-server-1  |   line: '2023',
sparkyfitness-server-1  |   routine: 'ExecConstraints'
sparkyfitness-server-1  | }
``

--- a/SparkyFitnessServer/services/mealService.js
+++ b/SparkyFitnessServer/services/mealService.js
@@ -409,6 +409,8 @@ async function createMealFromDiaryEntries(userId, date, mealType, mealName, desc
       name: mealName || defaultMealName,
       description: description,
       is_public: isPublic,
+      serving_size: 1.0,
+      serving_unit: 'serving',
       foods: mealFoods,
     };
 

